src/annotations/autolinks.ts

```diff
@@ -422,22 +427,30 @@ function ensureCachedRegex(ref: CacheableAutolinkReference, outputFormat: 'html'
 	// Regexes matches the ref prefix followed by a token (e.g. #1234)
 	if (outputFormat === 'markdown' && ref.messageMarkdownRegex == null) {
 		// Extra `\\\\` in `\\\\\\[` is because the markdown is escaped
-		ref.messageMarkdownRegex = new RegExp(
-			`(?<=^|\\s|\\(|\\\\\\[)(${escapeRegex(encodeHtmlWeak(escapeMarkdown(ref.prefix)))}(${
-				ref.alphanumeric ? '\\w' : '\\d'
-			}+))\\b`,
-			ref.ignoreCase ? 'gi' : 'g',
-		);
+		ref.messageMarkdownRegex = ref.regex
+			? new RegExp(`((${ref.regex}))`, 'g')
```

Comment:
When parsing the markdown, this is NOT working properly when escaped characters are found. Example:

The below regex:

```
(JIRA-[0-9]+)
```

Does NOT match a commit message with `JIRA-1234` because when escaped for Markdown, the text is:

```
JIRA\\-1234
```

I am looking for help to fix this issue.

Reply:
I thought on this for many days already and I still could not figure out a clean solution for this. I guess it's impossible to encode only literals of a regex string. Any help/feedback is welcome.